File: simplex-core/src/main/java/org/storck/simplex/model/Finalize.java
package org.storck.simplex.model;

import java.util.Objects;

/**
 * Message indicating that the iteration is finalized.
 *
 * @param playerId the ID of the player that sent the message
 * @param iteration the iteration number to finalizeMsg
 */
public record Finalize(String playerId, int iteration) {

    /**
     * Create the finalizeMsg message for the iteration.
     *
     * @param playerId the ID of the player that declares the iteration finalized
     * @param iteration the iteration number
     */
    public Finalize {
        Objects.requireNonNull(playerId);
    }
}

File: simplex-messages/src/main/scala/org/storck/simplex/message/Finalize.scala
package org.storck.simplex.message

/**
 * Message indicating that the iteration is finalized.
 *
 * @param playerId the ID of the player that sent the message
 * @param iterationNumber the iteration number to finalizeMsg
 */
case class Finalize(playerId: String, iterationNumber: Int) extends ProtocolIterationMessage

object Finalize {
  /**
   * Create the finalizeMsg message for the iteration.
   *
   * @param playerId the ID of the player that declares the iteration finalized
   * @param iterationNumber the iteration number
   */
  def apply(playerId: String, iterationNumber: Int): Finalize =
    new Finalize(playerId, iterationNumber)
}